src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -182,15 +191,24 @@ public override void Return(T[] array, bool clearArray = false)
                 {
                     t_tlsBuckets = tlsBuckets = new T[NumBuckets][];
                     tlsBuckets[bucketIndex] = array;
+                    if (s_TrimBuffers)
+                    {
+                        s_AllTlsBuckets.Add(tlsBuckets, null);
+                        if (Interlocked.Exchange(ref _callbackCreated, 1) != 1)
+                        {
+                            Gen2GcCallback.Register(Gen2GcCallbackFunc, this);
```

Comment:
I may have already asked this, but we're now going to end up with one of these Gen2 callback object things per T.  We decided that's ok?

Reply:
GetGC2Callback is pretty efficient (it holds just the callback and (weak) pointer to 'this'.  The main 'overhead' is that the object is finalizable.   You could imagine fixing that by sharing the finalizable object, but that is something that arguably should be done in the implementation of Gen2GcCallback.   

It think it is OK to wait in see (if it is a problem we can fix it).  

Also we may ditch Gen2GCCallback (since we really want to do this as we APPROACH a gen2 GC not after it), which is another reason not to invest alot of effort in it now.